minichain/pow.py

```
@@ -55,10 +55,14 @@ def mine_block(
         header_dict["nonce"] = local_nonce
         block_hash = calculate_hash(header_dict)
 
+        #check difficulty target 
+        target = "0" * difficulty
+
         # Check difficulty target
         if block_hash.startswith(target):
             block.nonce = local_nonce  # Assign only on success
             block.hash = block_hash
+            elapsed_time = time.monotonic() - start_time
```

Ruff (0.15.2)

[error] Local variable `elapsed_time` is assigned to but never used

Remove assignment to unused variable `elapsed_time`

(F841)